server/http/handlers/service.go

```diff
@@ -95,7 +127,10 @@ func (s *service) Write(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := s.writeBatch(data); err != nil {
+	writeBatchStart := s.nowFn()
+	err = s.writeBatch(data)
+	s.metrics.writeBatch.ReportSuccessOrError(err, s.nowFn().Sub(writeBatchStart))
```

Comment:
You can probably use `ReportSuccess` and `ReportError` since the error condition is checked below anyways.